### PR TITLE
Add libjson-rpc-cpp package

### PR DESCRIPTION
**Pull request by chfast**

### CODE REVIEW

File: cmake/projects/libjson-rpc-cpp/hunter.cmake [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+# List of versions:
+
+hunter_add_version(
+    PACKAGE_NAME
+    libjson-rpc-cpp
+    VERSION
+    "0.7.0-p0"
+    URL
+    "https://github.com/hunter-packages/libjson-rpc-cpp/archive/dbef3fafe4d9ce285befa78dbde55ab0f003a6c2.tar.gz"
```
Comment (ruslo) [inline cmake/projects/libjson-rpc-cpp/hunter.cmake]: I think we should create `v0.7.0-p0` release here.

File: cmake/projects/libjson-rpc-cpp/hunter.cmake [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    2e3ca04a9c55e7cee2541b498c70db606772c7e3
+)
+
+hunter_pick_scheme(DEFAULT url_sha1_cmake)
+
+hunter_cmake_args(
+    libjson-rpc-cpp
+    CMAKE_ARGS
+    BUILD_STATIC_LIBS=ON
+    BUILD_SHARED_LIBS=OFF
```
Comment (ruslo) [inline cmake/projects/libjson-rpc-cpp/hunter.cmake]: `BUILD_SHARED_LIBS=OFF` set by Hunter by default already

File: cmake/projects/libjson-rpc-cpp/hunter.cmake (hunk 1 above)
Comment (chfast) [inline cmake/projects/libjson-rpc-cpp/hunter.cmake]: Make sense.

Does `-p` means `-post`?

Comment (ruslo) [inline cmake/projects/libjson-rpc-cpp/hunter.cmake]: It means "patch". Also you can do `hunter-N`, like `v0.7.0-hunter-0`.

File: examples/libjson-rcp-cpp/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,12 @@
+cmake_minimum_required(VERSION 3.0)
```
Comment (ruslo) [inline examples/libjson-rcp-cpp/CMakeLists.txt]: Name of the directory should be `libjson-rpc-cpp`, not `libjson-rcp-cpp`.

Comment (chfast) [inline examples/libjson-rcp-cpp/CMakeLists.txt]: Fixed.

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.1693
* https://travis-ci.org/ingenue/hunter/builds/231114313?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
> Testing

Some tests failed

**Comment by chfast** [comment]
On Windows the problematic path is
```
C:\Users\chfast\Projects\hunter\hunter\_Base\xxxxxxx\46ba108\326fee5\Build\libjson-rpc-cpp\Build\libjson-rpc-cpp-Release-prefix\src\libjson-rpc-cpp-Release-build\src\jsonrpccpp\jsonrpcclientStatic.dir\Release\jsonrpcc.970C2D84.tlog\jsonrpccommonStatic.lastbuildstate
```

I can workaround it probably, but there is too much redundant information there. For sure you can shorter the `libjson-rpc-cpp-Release-prefix` to `Release`. You should take a look there.

**Comment by ruslo** [comment]
> For sure you can shorter the `libjson-rpc-cpp-Release-prefix` to `Release`

Also next component `Build` can be just `B`, see layout:
* https://docs.hunter.sh/en/latest/reference/layouts/deployed.html

Related issue:
* https://github.com/ruslo/hunter/issues/121

Feel free to investigate what we can do here and send the patch. Please check before that it fix all your problems of course :)

Also it would be nice if somebody clarify the Windows 10 improvements about this error:
* https://visualstudio.uservoice.com/forums/121579-visual-studio-2015/suggestions/4954037-fix-260-character-file-name-length-limitation

Can it be helpful for us or not?

**Comment by chfast** [comment]
How can I re-run the tests?

File: cmake/projects/libjson-rpc-cpp/hunter.cmake [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    SHA1
+    313bb442d1abcb6ea2761586a6ac3664d27c2ef5
+)
+
+hunter_pick_scheme(DEFAULT url_sha1_cmake)
+
+if(UNIX)
+  # UNIX options. Disable all as they don't compile on macOS.
+  set(
+    unix_cmake_args
```
Comment (ruslo) [inline cmake/projects/libjson-rpc-cpp/hunter.cmake]: set it to empty string if it's not UNIX, just in case

Comment (ruslo) [inline cmake/projects/libjson-rpc-cpp/hunter.cmake]: Also it's better to use something like `_hunter_unix_cmake_args`

Comment (chfast) [inline cmake/projects/libjson-rpc-cpp/hunter.cmake]: Give me a second.

**Comment by chfast** [comment]
Can we test now?

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.1702
* https://travis-ci.org/ingenue/hunter/builds/232088285?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Windows:
```
           CMake Error at src/jsonrpccpp/cmake_install.cmake:80 (file):
             file INSTALL cannot find
             "C:/projects/hunter/_testing/Hunter/_Base/125f2f6/2f7e5a5/326fee5/Build/libjson-rpc-cpp/Source/win32-deps/include".
```

**Comment by chfast** [comment]
Yeah, I forgot it tries to install to some stupid place on Windows.

**Comment by chfast** [comment]
Windows seems to be fixed now.

4 Linux/macOS failures:
- Both Linux and macOS fails to build OpenSSL with TOOLCHAIN=analyse, however in different places.
- CURL fails to build for iOS:
```
CURL/Source/lib/strerror.c:32:6: error: "strerror_r MUST be either POSIX, glibc or vxworks-style"
```
- jsoncpp fails to build on macOS with libstdcxx. I think libstdcxx might be old on macOS and it does not have std::unique_ptr. jsoncpp tries to fallback to std::auto_ptr, but fails to detect the need for it in this exotic case. This can be probably fixed upstream. Do you have tests for jsoncpp for this toolchain?

**Comment by ruslo** [comment]
> Do you have tests for jsoncpp for this toolchain?

I don't see `libstdcxx` toolchain:
* https://github.com/ingenue/hunter/branches/all?utf8=%E2%9C%93&query=pkg.jsoncpp

With dependencies the logic is the same as with the package itself: either send patches with fixes or exclude toolchains from testing.

**Comment by chfast** [comment]
jsoncpp issue reported https://github.com/open-source-parsers/jsoncpp/issues/612.

But considering that clang + libstdcxx on macOS does not support C++11 fully, I would consider it not critical issue.

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.1708
* https://travis-ci.org/ingenue/hunter/builds/232384851?utm_source=github_status&utm_medium=notification

**Comment by chfast** [comment]
All passed.

**Comment by ruslo** [comment]
Uploading:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.1709
* https://travis-ci.org/ingenue/hunter/builds/232395753?utm_source=github_status&utm_medium=notification

**Comment by chfast** [comment]
Can you explain what are uploading failures about?

**Comment by ruslo** [comment]
I missed the fact that "libjson-rpc-cpp" is not cacheable :) Nevermind.

**Comment by chfast** [comment]
What are the requirements for it to be?

**Comment by ruslo** [comment]
It should be relocatable:
* https://docs.hunter.sh/en/latest/creating-new/cmake.html#important

**Pull request merged**